### DISCUSSION TITLE
"destination exists" on parent dataset

### DISCUSSION THREAD

**Discussion by pendor** [psy0rz/zfs_autobackup#247] [Q&A] [answered]
I have a pool that I want to keep sync'd between two servers.  I've completed the initial send with zfs_autobackup successfully except for one error about the root/parent dataset of the transfer existing IE:
```
! [Target] STDERR > cannot receive new filesystem stream: destination 'drowning/mirrors/reflection' exists
! [Target] STDERR > must specify -F to overwrite it"
```
That dataset is essentially empty.  There's no file content within it, but it's the parent of everything else in this backup.  All of the other child datasets have replicated successfully, and running zfs_autobackup again creates new snapshots as appropriate and does incremental transfers, all as expected.  It's just that root dataset every time that complains.

Is there any way to recover from this situation without doing a full delete everything & re-send?  (That would take a minute...)  Is there some common error I might have made that put me in this situation to begin with?  Slightly abbreviated verbose log from a session is below.

As a side note, I'm new to zfs_autobackup, and I really appreciate the simplicity and intuitiveness of its workflow.  I've gone through a bunch of snapshot/send/receive tools and always found myself scratching my head at some of their decisions and fighting against their assumptions.  zfs_autobackup fulfills the Principle of Least Surprise, at least for my expectations.  It does what I expect, has sensible defaults, and seems like it should be a pretty good basis for backups I can depend on to JustWork™.  Many thanks!

Logs from the session:
```
# zfs-autobackup --verbose --ssh-target=penpen \
	--compress --buffer 2m --clear-mountpoint \
	--snapshot-format='{}-%Y%m%d%H%M' penpen drowning/mirrors

zfs-autobackup v3.2 - (c)2022 E.H.Eefting (edwin@datux.nl)

 Selecting dataset property : autobackup:penpen
 Snapshot format            : penpen-%Y%m%d%H%M
 Timezone                   : Local
 Hold name                  : zfs_autobackup:penpen

 #### Source settings
 [Source] Keep the last 10 snapshots.
 [Source] Keep every 1 day, delete after 1 week.
 [Source] Keep every 1 week, delete after 1 month.
 [Source] Keep every 1 month, delete after 1 year.

 #### Selecting
 [Source] reflection: Selected
 [Source] reflection/Servers: Selected
… [Lots more datasets] ...

 #### Snapshotting
 [Source] reflection: No changes since penpen-202403101621
 [Source] reflection/Servers: No changes since penpen-202403101621
… [Lots more datasets] ...

  #### Target settings
 [Target] SSH to: penpen
 [Target] Keep the last 10 snapshots.
 [Target] Keep every 1 day, delete after 1 week.
 [Target] Keep every 1 week, delete after 1 month.
 [Target] Keep every 1 month, delete after 1 year.
 [Target] Receive datasets under: drowning/mirrors

 #### Synchronising
 [Source] zfs send buffer        : 2m
 [Source] zfs send compression   : zstdmt -3
 [Target] zfs recv decompression : zstdmt -dc
 [Target] zfs recv buffer        : 2m
 [Source] reflection: sending to drowning/mirrors/reflection
 [Target] drowning/mirrors/reflection@penpen-202403101621: receiving full
! [Target] STDERR > cannot receive new filesystem stream: destination 'drowning/mirrors/reflection' exists
! [Target] STDERR > must specify -F to overwrite it
! [Target] Command "ssh penpen 'zstdmt -dc | mbuffer -q -s128k -m2m | zfs recv -u -o canmount=noauto -v -s drowning/mirrors/reflection'" returned exit code 1 (valid codes: [0])
! [Source] reflection: FAILED: Last command returned error
 [Source] reflection/Servers: sending to drowning/mirrors/reflection/Servers
… [Lots more datasets] ...
! 1 dataset(s) failed!
```

**Comment by psy0rz** [comment, marked as answer]
you can easily recover by running zfs-autobackup with --force one time.

This will only overwrite the data in that dataset, and it will leave the child datasets alone.
